### PR TITLE
Consolidated_weekend_ts_report

### PR DESCRIPTION
**Pull request by RohitNalePUCSD**
Consolidated weekdays/weekend timesheet report.
1. Input: Month selection
2. Output: This report will list extra hrs worked on weekdays (more than 9hrs) and on weekends
    This report should have the names of all engineers.

### CODE REVIEW

**Comment by RohitNalePUCSD** [comment]
[Consolidated weekend time sheet report for given date range.](https://trello.com/c/Z5kMdxTB/79-consolidated-weekend-time-sheet-report-for-given-date-range)

**Comment by Sameer-Tilak** [comment]
Overall report looks good. Ask Swapnil also to do code review. In the meantime please send me this report's sample to check it and suggest any changes if required.

**Review by swapnilchincholkar — APPROVED** [review]

**Comment by Sameer-Tilak** [comment]
Overall report LGTM. It will be properly tested once we generate this report from production data.

File: app/services/timesheet_monthly_service.rb [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+  def timesheet_missing_users(records, missing_ts_users_id_set, str_date)
+    missing_ts_users_id_set.each do |user_id|
+      records[user_id][:'timesheet'][:"#{str_date}"] = 'NF'
+      records[user_id][:'no_of_ts_missing_days'] += 1
+    end
+  end
+
+  def get_users_on_leave(date)
+    LeaveApplication.leaves.where(
```
Comment (Sameer-Tilak) [inline app/services/timesheet_monthly_service.rb:177]: Ieave_type - WFH should not be considered as leave. Need this filter in this query.

Comment (RohitNalePUCSD) [reply app/services/timesheet_monthly_service.rb:177]: Yes, Already filtered out WFH in leave_type and any user who takes an optional holiday is considered as leave and follows the same rules.

**Comment by Sameer-Tilak** [comment]
LGTM, Please deploy

**Pull request closed without merging**